examples/energy-management-app/energy-management-common/water-heater/src/WhmDelegateImpl.cpp

```diff
@@ -273,61 +300,77 @@ Status WaterHeaterManagementDelegate::HandleCancelBoost()
  *
  *********************************************************************************/
 
+uint16_t WaterHeaterManagementDelegate::GetActiveTargetWaterTemperature() const
+{
+    // Determine the target temperature. If a boost command is in progress and has a mBoostTemporarySetpoint value use that as the
+    // target temperature.
+    // Note, in practise the actual heating is likely to be controlled by the thermostat's occupiedHeatingSetpoint most of the
+    // time, and the TemporarySetpoint (if not null) would be overiding the thermostat's occupiedHeatingSetpoint.
+    // However, this code doesn't rely upon the thermostat cluster.
+    uint16_t targetTemperature = (mBoostState == BoostStateEnum::kActive && mBoostTemporarySetpoint.HasValue())
+        ? static_cast<uint16_t>(mBoostTemporarySetpoint.Value())
```

Comment:
Why is mBoostTemporarySetpoint a signed int?  And if it _is_ and can be negative, this cast will do really bad things, no?

Reply:
temporarySetpoint in the spec is signed to match the type used by thermostat.
This is copied from the BoostCommand (and used to send BoostStarted events).

So I think this should be signed. The question is can the others become signed as well... I think they can - so have made that change.